### PR TITLE
Fix #2133 allowed customization of crs list

### LINKED ISSUE #2133: Provide a way to whitelist the allowedCRS
Let's make it configurable from the mousePosition plugin config

### PR DESCRIPTION
**Pull request by MV88**
#2133

### CODE REVIEW

File: web/client/product/app.jsx [hunk 1]
```diff
@@ -12,10 +12,14 @@ const LocaleUtils = require('../utils/LocaleUtils');
 
 const startApp = () => {
     const ConfigUtils = require('../utils/ConfigUtils');
-
+    const CoordinatesUtils = require('../utils/CoordinatesUtils');
+    const projectionsDefs = require('./utils/projectionsDefs');
+    projectionsDefs().forEach((proj) => {
+        window.proj4.defs(proj.code, proj.def);
+        CoordinatesUtils.addProjections(proj.code, proj.extent, proj.worldExtent);
```
Comment (offtherailz) [inline web/client/product/app.jsx]: This should be in Openlayers map tool. Is not properly a CoordinateUtil, so move the internal code inside OL map.

File: web/client/product/app.jsx [hunk 2]
```diff
@@ -12,10 +12,14 @@ const LocaleUtils = require('../utils/LocaleUtils');
 
 const startApp = () => {
     const ConfigUtils = require('../utils/ConfigUtils');
-
+    const CoordinatesUtils = require('../utils/CoordinatesUtils');
+    const projectionsDefs = require('./utils/projectionsDefs');
+    projectionsDefs().forEach((proj) => {
+        window.proj4.defs(proj.code, proj.def);
```
Comment (offtherailz) [inline web/client/product/app.jsx]: You should read localConfig, so move this part forward in initialization chain.

File: web/client/utils/CoordinatesUtils.js [hunk 3]
```diff
@@ -247,6 +248,15 @@ const CoordinatesUtils = {
     isAllowedSRS(srs, allowedSRS) {
         return allowedSRS[CoordinatesUtils.getCompatibleSRS(srs, allowedSRS)];
     },
+    addProjections: function(code, extent, worldExtent) {
```
Comment (offtherailz) [inline web/client/utils/CoordinatesUtils.js]: move it in OL specific part

File: web/client/utils/projectionDefs.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017, GeoSolutions Sas.
```
Comment (offtherailz) [inline web/client/utils/projectionDefs.js]: remove and use localConfig

File: web/client/localConfig.json [hunk 5]
```diff
@@ -155,7 +155,12 @@
             }, {
                 "name": "MousePosition",
                 "cfg": {
-                    "id": "mapstore-mouseposition-mobile"
+                    "id": "mapstore-mouseposition-mobile",
+                    "editCRS": true,
+                    "showLabels": true,
+                    "showToggle": true,
+                    "filterAllowedCRS": ["EPSG:4326", "EPSG:3857", "EPSG:4629", "EPSG:102113"],
```
Comment (offtherailz) [inline web/client/localConfig.json]: Only 4326 and 3857

File: web/client/components/app/StandardApp.jsx [hunk 6]
```diff
@@ -47,6 +47,15 @@ class StandardApp extends React.Component {
         store: null
     };
 
+    addProjDefinitions(config) {
+        if (config.projectionDefs && config.projectionDefs.length) {
+            config.projectionDefs.forEach((proj) => {
+                window.proj4.defs(proj.code, proj.def);
```
Comment (mbarto) [inline web/client/components/app/StandardApp.jsx]: You should be able to require proj4 from web/client/libs as usual and use that one instead of window.proj4

File: web/client/components/app/StandardApp.jsx [hunk 7]
```diff
@@ -55,9 +64,11 @@ class StandardApp extends React.Component {
                     require('intl/locale-data/jsonp/en.js');
                     require('intl/locale-data/jsonp/it.js');
                     this.init(config);
+                    this.addProjDefinitions(config);
```
Comment (mbarto) [inline web/client/components/app/StandardApp.jsx]: Can we move this into init, so that we don't need to call it in two different places?

File: web/client/components/app/StandardApp.jsx [hunk 8]
```diff
@@ -55,9 +64,11 @@ class StandardApp extends React.Component {
...
                 });
             } else {
                 this.init(config);
+                this.addProjDefinitions(config);
```
Comment (mbarto) [inline web/client/components/app/StandardApp.jsx]: See above

File: web/client/localConfig.json [hunk 9]
```diff
@@ -30,6 +30,13 @@
     "monitorState": [
       {"name": "routing", "path": "routing.location.pathname"},
       {"name": "featuregridmode", "path": "featuregrid.mode"}],
+    "projectionDefs": [
```
Comment (mbarto) [inline web/client/localConfig.json]: I won't add any projection by default (why EPSG:3003 and not EPSG:3035?)

File: web/client/plugins/MousePosition.jsx [hunk 10]
```diff
@@ -7,7 +7,7 @@
  */
 const React = require('react');
 
-const {connect} = require('../utils/PluginsUtils');
+const {connect} = require('react-redux');
```
Comment (mbarto) [inline web/client/plugins/MousePosition.jsx]: Why did you change it?

Comment (MV88) [reply web/client/plugins/MousePosition.jsx]: i just restored it as it was before..

File: web/client/localConfig.json (hunk 9 above)
Comment (MV88) [reply web/client/localConfig.json]: sure, this was an intermediate pr. i'll remove it.

File: web/client/components/app/StandardApp.jsx (hunk 7 above)
Comment (MV88) [reply web/client/components/app/StandardApp.jsx]: i'll put it in the init method then

File: web/client/components/app/StandardApp.jsx (hunk 6 above)
Comment (MV88) [reply web/client/components/app/StandardApp.jsx]: 👍

File: web/client/plugins/MousePosition.jsx [hunk 11]
```diff
@@ -35,13 +35,15 @@ const getDesiredPosition = (map, mousePosition, mapInfo) => {
...
-], (map, mousePosition, mapInfo) => ({
+], (state, map, mousePosition, mapInfo) => ({
     enabled: mousePosition.enabled,
     mousePosition: getDesiredPosition(map, mousePosition, mapInfo),
-    crs: mousePosition.crs || map && map.projection || 'EPSG:3857'
+    projectionDefs: projectionDefsSelector(state),
+    crs: mousePosition.crs || 'EPSG:4326'
 }));
 
 const MousePositionButton = connect((state) => ({
```
Comment (offtherailz) [inline web/client/plugins/MousePosition.jsx]: document new available props
